src/druidry/data_source.py

```diff
@@ -322,6 +322,13 @@ def combine_filter(f):
         return OrFilter(fields=filters)
 
 
+def negate_filter(f):
+    filter = translate_filter(f['filter'])
```

Comment:
please use a different variable name like `f` as `filter` is a reserved keyword in python

Reply:
resolved